bundles/org.openhab.binding.tsmart/src/main/java/org/openhab/binding/tsmart/internal/TSmartHandler.java

```diff
@@ -0,0 +1,262 @@
+import org.openhab.core.library.unit.SIUnits;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.types.State;
+import org.openhab.core.types.UnDefType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link TSmartHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author James Melville - Initial contribution
+ */
+@NonNullByDefault
+public class TSmartHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(TSmartHandler.class);
+    private @Nullable TSmartConfiguration config;
+
+    private volatile int unRespondedRequests = 0;
+    private @Nullable InetAddress destAddr;
+
+    private @Nullable ScheduledFuture<?> statusRefreshJob;
+    private boolean propertiesInitializedSuccessfully = false;
+
+    public TSmartHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            requestStatusRefresh();
+        } else {
+            switch (channelUID.getId()) {
+                case CHANNEL_POWER:
+                    if (command instanceof OnOffType) {
+                        setPower(OnOffType.ON.equals(command));
+                    }
+                    break;
+                case CHANNEL_SETPOINT:
+                    if (command instanceof QuantityType<?> quantityCommand) {
+                        setSetpoint(quantityCommand);
+                    }
+                    break;
+                case CHANNEL_MODE:
+                    if (command instanceof StringType stringCommand) {
+                        setMode(stringCommand);
+                    }
+                    break;
+            }
+        }
+    }
+
+    /**
+     * Initialise ThingHandler - start listening for UDP responses,
+     * and sending status requests on the configured frequency.
+     */
+    @Override
+    public void initialize() {
+        config = getConfigAs(TSmartConfiguration.class);
+        TSmartConfiguration thisConfig = config;
+        if (thisConfig == null || thisConfig.hostname.isBlank() || thisConfig.refreshInterval <= 0) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR);
+        } else {
+            updateStatus(ThingStatus.UNKNOWN);
+            statusRefreshJob = scheduler.scheduleWithFixedDelay(this::requestStatusRefresh, 0,
+                    thisConfig.refreshInterval, TimeUnit.SECONDS);
+        }
+    }
+
+    private synchronized boolean initializeProperties() {
+        if (propertiesInitializedSuccessfully) {
```

Comment:
The method name and instance variable name seems a bit misleading. As I'm reading the method it is responsible for subscribing to UDP events, not initializing properties?

Reply:
Agree, the code is resolving the hostname (if a hostname and not plain IP, and creating a UDP listener / adding an address to listen to). Will resolve